file/src/main/scala/akka/stream/alpakka/file/impl/archive/TarArchiveHeader.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.file.impl.archive
+
+import java.lang.Long.toOctalString
+import java.time.Instant
+
+import akka.annotation.InternalApi
+import akka.util.ByteString
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[file] final class TarballHeader(val filePath: String,
+                                                     val size: Long,
+                                                     val lastModification: Instant = Instant.now) {
+
+  private val filePathSegments: Array[String] = filePath.split("/")
+  private val filePathPrefix = Option(filePathSegments.init).filter(_.nonEmpty).map(_.mkString("/"))
+  private val filePathName = filePathSegments.last
+
+  require(
+    filePathPrefix.forall(fnp => fnp.length > 0 && fnp.length <= 154),
+    "File path prefix must be between 1 and 154 characters long"
+  )
+  require(filePathName.length > 0 && filePathName.length <= 99,
+          "File path name must be between 1 and 99 characters long")
+
```

Comment:
This validation could be moved to `ArchiveMetadataWithSize` so that it can fail as early as possible.

Reply:
@ennru Was thinking about that too. But since this a highly TAR file specific constraint, I would have to name the class smth like `TarArchiveMetadata` and that looked bad next to `ArchvieMetadata`, which is only used for the ZIP archive. And I did not want to rename `ArchiveMetadata` -> `ZipArchiveMetadata`.

So I see three alternatives:

1. Stick with what is in this PR
2. Do what you suggested, but then rename
    a. `ArchiveMetadata` -> `ZipArchiveMetadata` and
    b. `ArchiveMetadataWithSize` -> `TarArchiveMetadata`

What do you prefer?

Reply:
Making these independent of each other seems best.
So make it `TarArchiveMetadata` and leave the existing `ArchiveMetadata` unchanged.